js/app/controllers/calcontroller.js

```diff
@@ -233,9 +233,24 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 							createAndRenderEvent(result.calendar, result.vevent.data, view.start, view.end, $scope.defaulttimezone);
 						}
 					}).catch(function(reason) {
-						if (reason === 'delete') {
-							deleteAndRemoveEvent(vevent, fcEvent);
+						if (reason !== 'delete') {
+							return true;
 						}
+
+						fc.elm.fullCalendar('removeEvents', fcEvent.id);
+						fcEvent.delete().then(function() {
+							return VEventService.delete(vevent).catch(function() {
+								throw new Error();
+							});
+						}).catch(function() {
+							const start = view.start.subtract(1, 'day');
+							const end = view.end.add(1, 'day');
+
+							const eventsToRender = vevent.getFcEvent(start, end, $scope.defaulttimezone);
+							angular.forEach(eventsToRender, function (event) {
+								fc.elm.fullCalendar('renderEvent', event);
+							});
+						});
```

Comment:
Good catch!

I tried using refetchEventSources, but I think it's a bit problematic. The event takes at least one second to reappear, because the browser has to send a CalDAV request to the server.
It would be way better, if the event would reappear immediately.